File: scala_akka/src/main/scala/com/e2edour/chat/server/SocketIOServerLauncher.java
package com.e2edour.chat.server;

import com.corundumstudio.socketio.*;
import com.corundumstudio.socketio.listener.ConnectListener;
import com.sun.xml.internal.bind.v2.runtime.output.SAXOutput;

/**
 * @author King
 * @version 2016/3/25
 */
public class SocketIOServerLauncher {

    private SocketIOServer server;

    public SocketIOServerLauncher() {
        Configuration config = new Configuration();
        config.setHostname("0.0.0.0");
        config.setPort(9979);
        config.setAckMode(AckMode.MANUAL);
        config.setTransports(Transport.WEBSOCKET);
        server = new SocketIOServer(config);
        System.out.println("初始化成功,你信吗？");
    }

    public void start() {
        server.start();
        server.addConnectListener(new ConnectListener() {
            public void onConnect(SocketIOClient client) {
                System.out.println("好像是客户端连进来了啊");
            }
        });
    }


}

File: scala_akka/src/main/scala/com/e2edour/chat/server/SocketIOServerLauncher.scala
package com.e2edour.chat.server

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.serializer.SerializerFeature
import com.corundumstudio.socketio._
import com.corundumstudio.socketio.listener.ConnectListener
import com.e2edour.chat.bean.{CustomerList, Room, User}

/**
  * 弄着玩
  *
  * @author King
  * @version 2016/3/28
  */
class SocketIOServerLauncher {

  val socketServer: SocketIOServer = new SocketIOServer(config);

  def config: Configuration = {
    val config: Configuration = new Configuration
    config.setHostname("0.0.0.0")
    config.setPort(8888)
    config.setAckMode(AckMode.MANUAL)
    config.setTransports(Transport.WEBSOCKET)
    config
  }

  def start = {
    socketServer.start
    socketServer.addConnectListener(new ConnectListener() {
      def onConnect(client: SocketIOClient) {
        val id = client.getHandshakeData.getSingleUrlParam("id")
        val user =User(id)
        val roomPath = "/user/private/"+user.id
        val room =Room(roomPath)
        client.set("room", room)
        client.set("user", user)
        client.joinRoom(roomPath)
      }
    })
  }

  def getServer: SocketIOServer = {
    socketServer
  }
}
